### PR TITLE
[WIP] Custom url

### PR DESCRIPTION
**Pull request by Th3Mouk**
https://github.com/sonata-project/SonataPageBundle/issues/502

### CODE REVIEW

**Comment by Th3Mouk** [comment]
@OskarStark can you help me I don't understand why test failed, seems getUrlFromPage in `Page.php` never called. :disappointed:

**Comment by OskarStark** [comment]
puh, don't know, the code looks ok to me

File: Route/CmsPageRouter.php [hunk 1]
```diff
@@ -305,18 +305,6 @@ protected function getPageByPageAlias($alias)
     }
 
     /**
-     * Returns the Url from a Page object.
-     *
-     * @param PageInterface $page
-     *
-     * @return string
-     */
-    protected function getUrlFromPage(PageInterface $page)
```
Comment (core23) [inline Route/CmsPageRouter.php]: This will be a BC break. You should add a deprecation notice instead.

Comment (Th3Mouk) [inline Route/CmsPageRouter.php]: Ok no problem did you know why the tests didn't pass ?

**Comment by Th3Mouk** [comment]
@Soullivaneuh can i have your help please. The method `$page->getUrlFromPage();` at line 204 in `CmsPageRouter.php` isn't called i'm beggining in testing thanks

File: Model/PageInterface.php [hunk 2]
```diff
@@ -151,6 +151,13 @@ public function setCustomUrl($customUrl);
     public function getCustomUrl();
 
     /**
+     * Get url of the page.
+     *
+     * @return string
+     */
+    public function getUrlFromPage();
```
Comment (core23) [inline Model/PageInterface.php]: Adding new stuff to an interface is a BC break

Comment (Th3Mouk) [inline Model/PageInterface.php]: Ok thanks ! Have you an idea for testing problem ?

**Comment by meyerbaptiste** [comment]
@Th3Mouk, tests must be updated. You need to mock your `Page::getUrlFromPage()` method, see [CmsPageRouterTest.php#L171](https://github.com/Th3Mouk/SonataPageBundle/blob/custom-url/Tests/Route/CmsPageRouterTest.php#L171).

**Comment by OskarStark** [comment]
thank you  for the help @meyerbaptiste

**Comment by OskarStark** [comment]
@Th3Mouk can you please rebase? we changed the travis conf.

@Th3Mouk does the comment of @meyerbaptiste helped?

**Comment by Th3Mouk** [comment]
@OskarStark i will check this end of week thx

**Comment by OskarStark** [comment]
thank you very much :+1:

**Comment by OskarStark** [comment]
@Soullivaneuh why did you close this PR?

**Comment by Soullivaneuh** [comment]
@OskarStark PR is auto-closed because the target branch was deleted.

@Th3Mouk Can you please reopen it on the right branch? :+1:

**Comment by OskarStark** [comment]
> @OskarStark PR is auto-closed because the target branch was deleted.

Ok, because there were no auto message, which describes it. Thanks for the feedback!

**Pull request closed without merging**